<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Michael Zietz" />
  <meta name="dcterms.date" content="2018-12-19" />
  <meta name="keywords" content="markdown, publishing, manubot" />
  <title>Network Methods for Multiomic Data Integration</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Network Methods for Multiomic Data Integration</h1>
</header>
<p><small><em> This manuscript (<a href="https://zietzm.github.io/integration-review/v/8a8669d23996ba12de8c189db345c09fc3eefa7a/">permalink</a>) was automatically generated from <a href="https://github.com/zietzm/integration-review/tree/8a8669d23996ba12de8c189db345c09fc3eefa7a">zietzm/integration-review@8a8669d</a> on December 19, 2018. </em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><strong>Michael Zietz</strong><br> <img src="images/orcid.svg" alt="ORCID icon" width="13" height="13" /> <a href="https://orcid.org/0000-0003-0539-630X">0000-0003-0539-630X</a> · <img src="images/github.svg" alt="GitHub icon" width="13" height="13" /> <a href="https://github.com/zietzm">zietzm</a><br> <small> Department of Physics, Department of Systems Pharmacology and Translational Therapeutics, University of Pennsylvania </small></li>
</ul>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>The age of abundant data has arrived. As the costs to generate and distribute omics-scale data have plummeted in the last two decades, previously impossible analysis is now reality. Through the integration of many types of omics data, diseases can increasingly be understood on their own terms: as highly complex, self-interacting, nonlinear biological systems. This review seeks to describe some of the many promising methods to simultaneously leverage multiple data modalities. Specifically, we examine the application of network methods as powerful, extensible, and natural ways to deal with multiple streams of information about a large number of entities. Significant progress has been made in the past two decades in unsupervised methods for data integration. While unsupervised methods can be extremely beneficial to supervised analyses, such applications remain an area of active and vibrant research. We conclude the review by exploring some of the potential avenues for future research and discovery.</p>
<h2 id="introduction">Introduction</h2>
<p>In the distant past, medical knowledge advanced toward the goal of finding treatments for every condition, defined very loosely. Once a treatment was found, the mystery disappeared and the disease had essentially been solved. Massive improvements have been made in the quality of medicine and the quantity of medical knowledge under this paradigm. At the current stage, however, diminishing returns can be achieved by simply trying to find a treatment for a broadly-defined disease. Two major avenues for future improvement stand out. First, the implementation of knowledge in real-world clinical settings. Despite the immensity of hard-gotten medical information available, patients continue to die from curable diseases and medical errors. Without putting into practice known ways to improve medicine, future research will fail to achieve its life-saving potential. Second, medical research must broaden its scope to encompass more information while zooming in to identify the specific biological foundations of disease, including the differences between individuals at the molecular level. Research toward the second goal is known as precision medicine, and this review covers a sub-field within precision medicine research.</p>
<h3 id="omics-and-precision-medicine">Omics and precision medicine</h3>
<p>Precision medicine entails custom-tailoring health care to individual patients based on their personal, clinical, and molecular profiles. The goal of precision medicine is to improve prevention, diagnosis, and treatment by accounting for the medically-relevant differences between individuals and groups. Toward this goal, the clinical value of omics data and omics-based discoveries has been demonstrated repeatedly. A brief overview of two sub-fields illustrates the possibilities for future research and the potential value of these discoveries.</p>
<p>Pharmacogenomics studies the genomic underpinnings of individual differences in drug response. Compared to the earlier-established field of pharmacogenetics, pharmacogenomics focuses on the simultaneous effects of many genes on drug metabolism and toxicity <span class="citation" data-cites="NzgwvuOF">[<a href="#ref-NzgwvuOF">1</a>]</span>. As of August 2018, the FDA lists 284 drugs with pharmacogenomic information included on the label <span class="citation" data-cites="1lcxSq1K">[<a href="#ref-1lcxSq1K">2</a>]</span>.</p>
<p>Deep phenotyping attempts to understand and fully classify the biological components underlying disease <span class="citation" data-cites="16REdtjMj">[<a href="#ref-16REdtjMj">3</a>]</span>. This task is made challenging by the fact that there are often not many well-understood links between genotype and phenotype, especially in the context of complex disease. Attention to varied patient information and its meaningful incorporation will require a new paradigm in medical diagnosis. To better understand diverse patient subtypes, consideration must be given to the patient’s full range of other diagnoses and symptoms, even if seemingly orthogonal <span class="citation" data-cites="gIKXoMmn">[<a href="#ref-gIKXoMmn">4</a>]</span>.</p>
<p>The ideal future for precision medicine would allow patients to be categorized into homogeneous groups for prevention, diagnosis, and treatment. That is, precision medicine seeks to uncover and account for the full range of individual variation that underlies differential disease susceptibility, prognosis, and treatment response. Whether in the form of specific clinical traits or detailed multiomic molecular attributes, precision medicine shows promise to overcome the heterogeneity of patients found in previously loosely-defined disease categories. If disease categorizations truly account for the totality of relevant factors, then patients within homogeneous categories can be treated systematically by the establishment of subtype-specific treatment regimes. Working toward such goals requires a broad approach, in which data from many different sources can be incorporated toward a more complete understanding of disease.</p>
<h3 id="multiomics">Multiomics</h3>
<p>While the clinical application of genomic data shows promise, biology cannot be fully explained by a single data type. Many diseases are affected in part by patients’ genomic profiles, though few diseases are based soley in genetics. The central dogma of molecular biology states that genetic information flows from nucleic acid to nucleic acid and from nucleic acid to protein. This dogma alone suggests that processes and information beyond the genetic sequence of a patient’s DNA will influence phenotype. <em>Multiomics</em> refers to the set of such biologically-relevant molecular data sources: genomics, epigenomics, transcriptomics, proteomics, metabolomics, and the organismal microbiome. Complex diseases may be most explainable when viewed through the lenses of multiple such data sources.</p>
<p>In recent years, the cost of omics data generation has fallen dramatically. A former target of $1,000 per genome sequenced now seems high as companies like Illumina promise a future of the, “$100 genome,” <span class="citation" data-cites="Ml77ulah">[<a href="#ref-Ml77ulah">5</a>]</span>. Such rapid technological improvements have recently made the creation of large-scale multiomic datasets cost-effective. The Cancer Genome Atlas (TCGA) is a large-scale project which has collected patient information and tumor samples from over 11,000 patients in the United States. Its data include clinical information about patients, sample metadata, and multiomic sample data including gene expression, SNP genotypes, copy number variation (CNV), DNA methylation, exon sequencing, and microRNA profiles <span class="citation" data-cites="AK17eOgD">[<a href="#ref-AK17eOgD">6</a>]</span>. Another project underway is the International Cancer Genome Consortium (ICGC), which seeks to incorporate data from TCGA and around 50 other projects to provide researchers access to a massive quantity of tumor omics data <span class="citation" data-cites="ynLCqVng">[<a href="#ref-ynLCqVng">7</a>]</span>. The proliferation and falling costs of next-generation sequencing technologies have allowed these and other large consortia to create massive data resources covering many modalities and diverse diseases. Decreasing costs of data have not, however, resulted in a decrease in the cost of data analysis. Significant future research is needed to deeply probe multiomic datasets in order to extract the data that will guide future diagnostics and treatments.</p>
<h3 id="data-integration">Data integration</h3>
<p>Different data modalities can contain distinct pieces of medically-relevant information. This has led researchers and clinicians to consider how multiple data modalities can together provide a more complete picture of disease. An obvious approach for prediction or classification is the simple combination (or concatenation) of data from multiple modalities. The issue with such approaches has to do with the data themselves. Omics data often contain a very large number of features relative to the number of samples. This leads to a low signal-to-noise ratio, which is only worsened by the concatenation of data from two or more different streams. An alternative and promising approach involves extracting information from data modalities individually before combining extracted data into a single, more cohesive representation with a relatively higher signal-to-noise ratio. Several such techniques have been published and a subset of them will be covered later in this review.</p>
<p>Data-integrative methods are advantageous because they allow researchers to discover connections which are attested by multiple sources of data. This advantage is extremely beneficial, as it allows integrative methods to achieve fewer false positives when compared to single-modality methods. Particularly in the age of high-throughput omics technologies, where data can be staggeringly abundant, false positives are a major problem and steps must very often be taken to reduce their influence.</p>
<p>Another advantage of methods which simultaneously consider multiple sources of data is their superior ability to uncover complex and emergent relationships. Data-integrative approaches to multiomic analysis are better suited to capture nonlinear relationships across data modalities than the simple combination of single-modality-derived conclusions. To better illustrate this point, consider the example biological system proposed by Ritchie et al. <span class="citation" data-cites="zR4nSSeT">[<a href="#ref-zR4nSSeT">8</a>]</span>. They imagine two competing hypotheses to explain a complex phenotype. First, so-called “Hypothesis A” is a linear system of explanation in which genomic variation leads to variation in gene expression, leading to variation in protein expression, leading to a certain phenotype. The authors propose that Hypothesis A would be best tested by step-wise progress and incremental data reduction/filtering. An alternative system, “Hypothesis B”, the nonlinear hypothesis, involves multi-interacting connections between genome, transcriptome, proteome, epigenome and phenotype. Hypothesis B, the authors suggest, would be better tested using methods that combine the relevant data modalities first, rather than filtering each type to those entities believed <em>a priori</em> to be relevant to the phenotype.</p>
<p>As an even simpler example, consider the exclusive or (XOR) function. It is impossible to learn the behavior of XOR while only able to modulate or observe one input. Only by observing both relevant inputs can the function be determined. Similarly, to completely understand complex and nonlinear biological systems, hidden variables must be unveiled and understood in the context of all the relevant data. This task is best achieved by simultaneously considering information from multiple sources, the goal of data integration methods.</p>
<h3 id="network-approaches-to-data-integration">Network approaches to data integration</h3>
<p>Networks are a logical way of representing many types of biological data. Nodes (or vertices) may represent biological or biomedical entities, while edges may represent a number of different connections or pairings between entities. Several graph-theoretic layers can be helpful in the construction and analysis of biological networks, including graph coloring, edge weighting, and the specification of edge direction. By including such additional layers of information, as well as the association of nodes and edges with biological metadata, networks can consolidate many types of information into a single, cohesive format.</p>
<p>This review focuses on network-based methods for data integration, though such methods are not the only ones which show promise. For further discussion of alternative approaches, several excellent reviews and comparisons are available <span class="citation" data-cites="zR4nSSeT 156UVA1sT lOKGIFIL lOKGIFIL">[<a href="#ref-lOKGIFIL">10</a>,<a href="#ref-lOKGIFIL">10</a>,<a href="#ref-zR4nSSeT">8</a>,<a href="#ref-156UVA1sT">9</a>]</span>.</p>
<h2 id="discussion">Discussion</h2>
<p>Machine learning techniques have broadly been classified into two disjoint categories, supervised and unsupervised.<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> While in reality the two categories represent endpoints of a continuum, this review will consider the only the binary discretum. Through such a lens, “supervised” refers to techniques which attempt to learn mappings between specified inputs and outputs. A simple example of a supervised learning technique is logistic regression, which attempts to assign binary labels to data points after being trained on labeled data. “Unsupervised” learning refers to techniques which identify characteristics of data without labels. A simple example of unsupervised learning is principal component analysis (PCA), which transforms observations onto a new set of coordinates that represent the axes of greatest variance. The discussion that follows is split between supervised and unsupervised methods. At present, greater advances appear to have been made in unsupervised methods, though supervised methods can be strengthened and enabled by first applying unsupervised data integration, feature selection, or de-noising.</p>
<h3 id="unsupervised-methods">Unsupervised methods</h3>
<p>A very useful application of unsupervised methods is the transformation of large noisy data into a useful, de-noised format which can used for supervised machine learning. For example, Greene et al. <span class="citation" data-cites="CVF61Un5">[<a href="#ref-CVF61Un5">11</a>]</span> constructed tissue-specific functional gene networks using data from over 14,000 experiments. By integrating tissue-specific expression information in a Bayesian framework, the authors constructed networks of tissue-specific gene functional relationships, with edge weights representing the posterior probability of a functional relationship. These edges weights were determined by a naive Bayes classifier that was trained to predict positive functional relationships based on tissue-specific data. A key insight by the authors was that tissue-specific networks represent a data-driven method for the analysis of noisy GWAS data. The corresponding method is outlined later in this review.</p>
<p>Another data-integrative application of functional networks was proposed by Sehgal et al. <span class="citation" data-cites="dMXD2313">[<a href="#ref-dMXD2313">12</a>]</span>, who developed a methodology called the Robust Selection Algorithm (RSA) to identify micro-RNAs (miRNA) which are oncogenic or tumor-suppressing. By incorporating functional networks of miRNA-mRNA data, the authors were able to account for non-linear relationships between molecular markers and patient outcomes. Using data from TCGA, RSA was able to identify a number of both onco-miRNA and tumor-suppressor miRNA with relevance for the prediction of survival times in various cancers. To generate functional networks, the authors employed the NetWalker <span class="citation" data-cites="19R0rfh1g">[<a href="#ref-19R0rfh1g">13</a>]</span> suite of software, a self-described, “one-stop platform,” for network-based data methods for integrative genomics.</p>
<p>A common goal of network-based systems biology is the identification of functionally-related modules (or sub-networks) within larger networks of genes or proteins. Ideker et al. <span class="citation" data-cites="sL4vH9fY">[<a href="#ref-sL4vH9fY">14</a>]</span> developed an approach to integrate protein-protein and protein-DNA interactions with mRNA expression data to identify “active” subnetworks. Defined as, “connected sets of genes with unexpectedly high levels of differential expression,” active subnetworks were identified by finding groups of nodes with significant differential expression, the <em>collective</em> expression of which was compared to randomly chosen subnetworks. Having scored subnetworks, the authors’ method then finds the highest-scoring subnetworks using simulated annealing. Subsequent research in module network inference has expanded the methods’ capabilities to most common types of omic data. Lemon-Tree, due to Bonnet et al. <span class="citation" data-cites="v5FVh9X6">[<a href="#ref-v5FVh9X6">15</a>]</span>, is an open-source software package that encompasses many validated algorithms and novel approaches for module network inference.</p>
<p>iCluster <span class="citation" data-cites="cMcCaN4S">[<a href="#ref-cMcCaN4S">16</a>]</span> is a joint latent variable model, which attempts to estimate unobserved tumor subtypes using available multiomic data such as CNV, DNA methylation, and mRNA expression. The method estimates tumor subtypes by maximizing a set of linear equations using maximum likelihood and the expectation-maximization algorithm. Applying the method to breast cancer, the authors were able to identify four disease subtypes, recovering well-known clinical subtypes, as well as identifying potentially novel subtypes of the disease.</p>
<p>Network-based methods for biomedical data integration are not limited to representations in which nodes represent genes, proteins, RNA molecules, or other molecular entities. Similarity Network Fusion (SNF), a method due to Wang, et al. <span class="citation" data-cites="1Hk7MyXmO">[<a href="#ref-1Hk7MyXmO">17</a>]</span>, uses patient similarity networks to cluster patients into discrete and clinically-meaningful groups. In these networks, nodes represent patients and the edges connecting patients represent their pairwise similarity. For each additional data modality, SNF creates a similarity network based on the patients’ molecular profiles. The constructed modality-specific networks are then fused using an iterative linear algebra routine which is appropriate for studies of varied sample size and feature number, as well as for highly heterogeneous data sources. The authors applied SNF to three data modalities for a case study of glioblastoma multiforme (GBM), an aggressive brain tumor which is challenging to treat <span class="citation" data-cites="1Gtz7k2li">[<a href="#ref-1Gtz7k2li">18</a>]</span>. Previous data-integrative approaches had shown varied results, including the identification of a variable numbers of disease subtypes, depending on the data modality under investigation. Using SNF, the authors were able to recover known clinically-relevant disease subtypes and show that these subtypes correspond to very different survival prognoses. Moreover, by probing the network resulting from the SNF method, the authors were able to show that the majority of the edges were supported by two or more different data modalities, strengthening confidence in the validity of their integrative method.</p>
<p>Applying individual weights to networks is a natural next-step for the fusion of multiomic networks. In a separate application of SNF, Angione et al. <span class="citation" data-cites="zWyXFgao">[<a href="#ref-zWyXFgao">19</a>]</span> developed a network approach for bacterial growth and response, in which nodes represent growth conditions and edges represent the similarities between conditions. In a modification to the original SNF method, Angione et al. propose a weighted or “biased” SNF method in which layers themselves can be weighted according to the predictive quality of the various data. The authors state that this modification also increases the method’s robustness to distributions with high kurtosis. After consolidating the network into a single, global representation of the multi-modal data, the authors were able to predict <em>E. coli</em> metabolism under a range of possible growth conditions.</p>
<p>Mosca et al. <span class="citation" data-cites="sUdFrajH">[<a href="#ref-sUdFrajH">20</a>]</span> developed a multi-objective data fusion method, in which both component- and network-level estimators are optimized. As in the biased SNF method, networks themselves are weighted during the integration procedure. Unlike the previous work, though, the multi-objective method finds optimal weights for both networks and nodes, thus leaving fewer hyperparameters needing user-given specification. A difficulty of the MO method lies in the definition of objective functions to be minimized. While the authors propose several (over-representation analysis statistics, functional class scoring à la GSEA, etc.), it is not clear which metric would be most appropriate for any given situation. Nonetheless, the method presented is an interesting and novel way to optimize the multiomic sets of networks that arise in integrative genomics.</p>
<p>Another key application for unsupervised data integration methods is the prioritization or ranking of the molecular causes of disease. Greene et al. <span class="citation" data-cites="CVF61Un5">[<a href="#ref-CVF61Un5">11</a>]</span> developed a method called NetWAS to re-prioritize genes with nominally-significant p-values from standard GWAS methods. NetWAS uses tissue-specific functional networks as features when training a support vector machine (SVM) classifier to predict whether a gene was nominally significant in a GWAS study. After training, genes can be re-ranked according to their distance from the SVM decision hyperplane. The rationale behind this method is that the SVM is able to capture some characteristics of the data which lead to significant gene enrichment, while genes with spurious p-values will not share the same functional characteristics. One of the pivotal innovations of NetWAS is the ability to re-prioritize or cross-prioritize genes from distinct GWAS studies in a method that does not depend on <em>a priori</em> knowledge of the disease.</p>
<p>Aerts et al. <span class="citation" data-cites="dTwzcIZ4">[<a href="#ref-dTwzcIZ4">21</a>]</span> also developed a method to prioritize the genes that potentially underlie disease. Named Endeavor, the method ranks genes based on their similarity to genes known to be involved in the disease. After producing a ranking using each data modality available separately, the ranking lists are combined. In a later paper, De Bie et al. <span class="citation" data-cites="10RgFYeEF">[<a href="#ref-10RgFYeEF">22</a>]</span> improved upon the Endeavor methodology by devising a kernel-based method to rank genes within a single data modality. Both of these methods have the disadvantage of needing comparison data in which to contextualize unseen observations.</p>
<p>Overall, unsupervised data integration methods have made tremendous progress in the past two decades. These methods have shown promise for the generation of useful datasets, the data-driven identification of disease subtypes and patient clusters, as well as for the extraction of information pertinent to prediction tasks.</p>
<h3 id="supervised-methods">Supervised methods</h3>
<p>While supervised learning methods are methodologically quite distinct from unsupervised approaches, the underlying learning is fundamentally similar. In both cases, the goal is to develop computational models that are able to learn about the intrinsic qualities, connections, and variation patterns within high-dimensional, noisy data. Supervised learning then leverages this lower-level understanding to make explicit and testable predictions about observed data.</p>
<p>A classic application of supervised learning to networks is the task of edge prediction. Within networks of a single type, several common methods exist. A basic approach involves dropping several nodes from the network, computing walk-counts between pairs of nodes, then using these counts as features to predict the edges which were dropped. Superior methods have been devised, but few are immediately applicable to multi-modality datasets in a way that would produce meaningful and interpretable results. Another key application of supervised learning is the prediction of node sets that are related in some way or relevant to some other phenomenon. Among the most common applications is the prediction of genes that are relevant to a particular disease.</p>
<p>Himmelstein et al. <span class="citation" data-cites="WkPlH1ds">[<a href="#ref-WkPlH1ds">23</a>]</span> created a so-called “heterogeneous” network with nodes representing biomedical entities of many different types. Fundamentally, the authors’ data integration strategy is simply the joining of multiple types of relationship data (eg: genes, diseases, molecular functions, compounds) along sets of common vocabularies. After creating a multipartite network with over 40,000 nodes and 1,600,000 edges, the authors performed a number of prediction tasks, illustrating the power of integrative approaches. As features to their predictive models, the authors devised a novel metric of node pair interconnected-ness, termed the “degree-weighted path count,” (DWPC). Computed by traversing the network across multiple semantic types of paths (eg: Gene-Associates-Disease-treated by-Compound), DWPC allowed the authors to predict withheld high-confidence associations from the GWAS catalog at an area under the receiver operating curve (AUROC) of 0.83.</p>
<p>In a follow-up application <span class="citation" data-cites="O21tn8vf">[<a href="#ref-O21tn8vf">24</a>]</span> the authors utilized a similar method to predict candidates for drug repurposing within a much-enlarged network. Predicting treatments for epilepsy, the authors were able to recapture 23 of 25 (withheld) known anti-epileptic drugs. One of the advantages of the heterogeneous network method is that connections between any two types of biological entities in the network can be predicted using the same methodology. A drawback of this method is that gold-standard data must be obtained for the connection-type to be predicted, which could be challenging for certain types of biological connections.</p>
<p>Hypothesizing that many genes involved in cancer development may not be identified by conventional methods such as differential expression or mutation analysis, Ruffalo et al. <span class="citation" data-cites="Vvpiw3GG">[<a href="#ref-Vvpiw3GG">25</a>]</span> devised a method to identify these so-called, “Silent players.” Fundamentally, the method seeks to identify proteins involved in cancer by computing the proximity of all human proteins to products of genes which are differentially-expressed or mutated. “Proximity” in this context is computed by network propagation and results in two vectors (one for each of the input data modalities). These features are then used to predict the probability of the gene being involved in the disease of interest. While the method described in the paper integrates only mutation and differential expression data with protein-protein interaction networks, the method could, in principle, be applied to similar data modalities as well, provided suitable networks exist for the modalities of interest.</p>
<p>Supervised methods also lend themselves to the prediction of clinical outcomes. Kim et al. <span class="citation" data-cites="X7vddlEk">[<a href="#ref-X7vddlEk">26</a>]</span> used multiomic data to predict outcomes for several cancers. Their data integration method first created a patient similarity network for each data modality. Next, the individual networks were merged by finding the optimal coefficients for a linear combination of graphs such that an objective function involving the graph Laplacian was minimized.<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> This work preceded the publication of SNF, though it is similar in approach and in effect. After merging modality-specific networks into a global network of patient similarities, the authors predicted clinical outcomes using a nearest-neighbor approach.</p>
<p>Shin et al. <span class="citation" data-cites="aPrQxP1f">[<a href="#ref-aPrQxP1f">27</a>]</span> created an integrated network of proteins by finding the optimal linear combination of individual graph Laplacians. Importantly, they developed a method called graph sharpening, in which edges between labeled and unlabeled points are removed from the graph. Sharpening reduces the noise found in a graph by eliminating undesired edges. The authors found that this method, when applied in conjunction with network integration, affords the greatest improvement for the task of predicting whether proteins are members of a gene ontology (GO) category.</p>
<p>Other clinical events beyond the emergence of disease can be predicted as well. Mankoo et al. <span class="citation" data-cites="18IBB9pE2">[<a href="#ref-18IBB9pE2">28</a>]</span> developed a model with a focus on reducing the number of features used for prediction and increasing the biological interpretablility of outcomes. In the study, the authors sought to predict clinical outcomes for patients with serous ovarian cancer using multiomic data from TCGA. Their method filtered features by computing the Spearman rank correlations between data types and removing all features below chosen cut-offs. Overall, this procedure reduced the number of features from roughly 50,000 to under 200. Using the reduced feature set, the authors predicted discrete outcomes and continuous time to recurrence better than with any of the data types alone.</p>
<h2 id="conclusion">Conclusion</h2>
<p>In this review, we have described many of the major recent advances in network-based methods for multiomic data integration. Networks are natural representations of many types of biological and biomedical data, and they can be very effectively leveraged for integrative analyses.</p>
<p>Unsupervised methods attempt to integrate data for the purposes of feature reduction, de-noising, clustering, stratification, and future prediction methods. Another application which is particularly relevant to unsupervised network methods is the identification of functional modules (or sub-networks) that correspond to biologically-meaningful functional units. Significant conceptual advances include the creation of networks in which nodes have unconventional meaning, such as bacterial growth conditions or patients themselves. Unsupervised integrative analyses have also shown their power for the prioritization (or re-prioritization) of molecular entities according to their potential influence on phenotype.</p>
<p>An intriguing future direction of research involves the application of novel network methods to integrated omics networks. For example, Tan et al. <span class="citation" data-cites="Hlprh8TG">[<a href="#ref-Hlprh8TG">29</a>]</span> used an ensemble of unsupervised methods to extract molecular signatures from gene expression data. The authors employed an ensemble of denoising autoencoders to extract stable molecular signatures from the highly over-parametrized model. Similar ensemble-of-unsupervised approaches could be applied to integrate multi-modal omics data.</p>
<p>Supervised methods attempt to harness multifarious data to make biological or clinical predictions. While supervised methods lend themselves less naturally to the practice of data integration, many unsupervised methods return integrated data which is particularly amenable to supervised prediction. A major future advancement in this field will undoubtedly involve the association of outcome labels with multiomics data of a massive scale. Specifically, the integration of omics data with electronic medical records (EMR) could lead to the creation and availability of huge data sets alongside information about disease, treatment, and response. Such an integration is one of the most promising avenues for future work in this field, and it would lay the path for significant future discovery. Among the many types of research that such integration would allow, empirically derived “deep phenotypes” are some of the most exciting possibilities.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-NzgwvuOF">
<p>1. <strong>Pharmacogenomics: Translating Functional Genomics into Rational Therapeutics</strong><br />
W. E. Evans<br />
<em>Science</em> (1999-10-15) <a href="https://doi.org/cw248b">https://doi.org/cw248b</a><br />
DOI: <a href="https://doi.org/10.1126/science.286.5439.487">10.1126/science.286.5439.487</a></p>
</div>
<div id="ref-1lcxSq1K">
<p>2. <strong>Table of Pharmacogenomic Biomarkers in Drug Labeling</strong><br />
U.S. Food and Drug Administration<br />
<em>U.S. Food and Drug Administration</em> (2018-08-03) <a href="https://www.fda.gov/drugs/scienceresearch/ucm572698.htm">https://www.fda.gov/drugs/scienceresearch/ucm572698.htm</a></p>
</div>
<div id="ref-16REdtjMj">
<p>3. <strong>Deep phenotyping for precision medicine</strong><br />
Peter N. Robinson<br />
<em>Human Mutation</em> (2012-04-13) <a href="https://doi.org/gfpptq">https://doi.org/gfpptq</a><br />
DOI: <a href="https://doi.org/10.1002/humu.22080">10.1002/humu.22080</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22504886">22504886</a></p>
</div>
<div id="ref-gIKXoMmn">
<p>4. <strong>Deep phenotyping: The details of disease</strong><br />
Cathryn M. Delude<br />
<em>Nature</em> (2015-11) <a href="https://doi.org/gfpptr">https://doi.org/gfpptr</a><br />
DOI: <a href="https://doi.org/10.1038/527s14a">10.1038/527s14a</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26536218">26536218</a></p>
</div>
<div id="ref-Ml77ulah">
<p>5. <strong>Cheaper DNA sequencing unlocks secrets of rare diseases</strong><br />
Sarah Neville<br />
<em>Financial Times</em> (2018-03-05) <a href="https://ft.com/content/017a3a50-f6f1-11e7-a4c9-bbdefa4f210b">https://ft.com/content/017a3a50-f6f1-11e7-a4c9-bbdefa4f210b</a></p>
</div>
<div id="ref-AK17eOgD">
<p>6. <strong>The Cancer Genome Atlas Pan-Cancer analysis project</strong><br />
John N WeinsteinEric A Collisson, Gordon B Mills, Kenna R Mills Shaw, Brad A Ozenberger, Kyle Ellrott, Ilya Shmulevich, Chris Sander, Joshua M Stuart<br />
<em>Nature Genetics</em> (2013-10) <a href="https://doi.org/f3nt5c">https://doi.org/f3nt5c</a><br />
DOI: <a href="https://doi.org/10.1038/ng.2764">10.1038/ng.2764</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24071849">24071849</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3919969">PMC3919969</a></p>
</div>
<div id="ref-ynLCqVng">
<p>7. <strong>International network of cancer genome projects</strong><br />
Thomas J. Hudson (Chairperson), Warwick Anderson, Axel Aretz, Anna D. Barker, Cindy Bell, Rosa R. Bernabé, M. K. Bhan, Fabien Calvo, Iiro Eerola, Daniela S. Gerhard, … Huanming Yang<br />
<em>Nature</em> (2010-04-15) <a href="https://doi.org/cm9h2m">https://doi.org/cm9h2m</a><br />
DOI: <a href="https://doi.org/10.1038/nature08987">10.1038/nature08987</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/20393554">20393554</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2902243">PMC2902243</a></p>
</div>
<div id="ref-zR4nSSeT">
<p>8. <strong>Methods of integrating data to uncover genotype–phenotype interactions</strong><br />
Marylyn D. Ritchie, Emily R. Holzinger, Ruowang Li, Sarah A. Pendergrass, Dokyoon Kim<br />
<em>Nature Reviews Genetics</em> (2015-01-13) <a href="https://doi.org/bg9k">https://doi.org/bg9k</a><br />
DOI: <a href="https://doi.org/10.1038/nrg3868">10.1038/nrg3868</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25582081">25582081</a></p>
</div>
<div id="ref-156UVA1sT">
<p>9. <strong>More Is Better: Recent Progress in Multi-Omics Data Integration Methods</strong><br />
Sijia Huang, Kumardeep Chaudhary, Lana X. Garmire<br />
<em>Frontiers in Genetics</em> (2017-06-16) <a href="https://doi.org/gcz6m3">https://doi.org/gcz6m3</a><br />
DOI: <a href="https://doi.org/10.3389/fgene.2017.00084">10.3389/fgene.2017.00084</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28670325">28670325</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5472696">PMC5472696</a></p>
</div>
<div id="ref-lOKGIFIL">
<p>10. <strong>Methods for the integration of multi-omics data: mathematical aspects</strong><br />
Matteo Bersanelli, Ettore Mosca, Daniel Remondini, Enrico Giampieri, Claudia Sala, Gastone Castellani, Luciano Milanesi<br />
<em>BMC Bioinformatics</em> (2016-01-20) <a href="https://doi.org/gcpgct">https://doi.org/gcpgct</a><br />
DOI: <a href="https://doi.org/10.1186/s12859-015-0857-9">10.1186/s12859-015-0857-9</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26821531">26821531</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4959355">PMC4959355</a></p>
</div>
<div id="ref-CVF61Un5">
<p>11. <strong>Understanding multicellular function and disease with human tissue-specific networks</strong><br />
Casey S Greene, Arjun Krishnan, Aaron K Wong, Emanuela Ricciotti, Rene A Zelaya, Daniel S Himmelstein, Ran Zhang, Boris M Hartmann, Elena Zaslavsky, Stuart C Sealfon, … Olga G Troyanskaya<br />
<em>Nature Genetics</em> (2015-04-27) <a href="https://doi.org/f7dvkv">https://doi.org/f7dvkv</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3259">10.1038/ng.3259</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25915600">25915600</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4828725">PMC4828725</a></p>
</div>
<div id="ref-dMXD2313">
<p>12. <strong>Robust Selection Algorithm (RSA) for Multi-Omic Biomarker Discovery; Integration with Functional Network Analysis to Identify miRNA Regulated Pathways in Multiple Cancers</strong><br />
Vasudha Sehgal, Elena G. Seviour, Tyler J. Moss, Gordon B. Mills, Robert Azencott, Prahlad T. Ram<br />
<em>PLOS ONE</em> (2015-10-27) <a href="https://doi.org/f78knm">https://doi.org/f78knm</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pone.0140072">10.1371/journal.pone.0140072</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26505200">26505200</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4623517">PMC4623517</a></p>
</div>
<div id="ref-19R0rfh1g">
<p>13. <strong>NetWalker: a contextual network analysis tool for functional genomics</strong><br />
Kakajan Komurov, Serkan Dursun, Serkan Erdin, Prahlad T Ram<br />
<em>BMC Genomics</em> (2012) <a href="https://doi.org/gb3fdc">https://doi.org/gb3fdc</a><br />
DOI: <a href="https://doi.org/10.1186/1471-2164-13-282">10.1186/1471-2164-13-282</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22732065">22732065</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439272">PMC3439272</a></p>
</div>
<div id="ref-sL4vH9fY">
<p>14. <strong>Discovering regulatory and signalling circuits in molecular interaction networks</strong><br />
T. Ideker, O. Ozier, B. Schwikowski, A. F. Siegel<br />
<em>Bioinformatics</em> (2002-07-01) <a href="https://doi.org/cwkstn">https://doi.org/cwkstn</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/18.suppl_1.s233">10.1093/bioinformatics/18.suppl_1.s233</a></p>
</div>
<div id="ref-v5FVh9X6">
<p>15. <strong>Integrative Multi-omics Module Network Inference with Lemon-Tree</strong><br />
Eric Bonnet, Laurence Calzone, Tom Michoel<br />
<em>PLOS Computational Biology</em> (2015-02-13) <a href="https://doi.org/gcpgc3">https://doi.org/gcpgc3</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1003983">10.1371/journal.pcbi.1003983</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25679508">25679508</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4332478">PMC4332478</a></p>
</div>
<div id="ref-cMcCaN4S">
<p>16. <strong>Integrative clustering of multiple genomic data types using a joint latent variable model with application to breast and lung cancer subtype analysis</strong><br />
Ronglai Shen, Adam B. Olshen, Marc Ladanyi<br />
<em>Bioinformatics</em> (2009-09-16) <a href="https://doi.org/dvdz5s">https://doi.org/dvdz5s</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btp543">10.1093/bioinformatics/btp543</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/19759197">19759197</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2800366">PMC2800366</a></p>
</div>
<div id="ref-1Hk7MyXmO">
<p>17. <strong>Similarity network fusion for aggregating data types on a genomic scale</strong><br />
Bo Wang, Aziz M Mezlini, Feyyaz Demir, Marc Fiume, Zhuowen Tu, Michael Brudno, Benjamin Haibe-Kains, Anna Goldenberg<br />
<em>Nature Methods</em> (2014-01-26) <a href="https://doi.org/f5v9f5">https://doi.org/f5v9f5</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.2810">10.1038/nmeth.2810</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24464287">24464287</a></p>
</div>
<div id="ref-1Gtz7k2li">
<p>18. <strong>Glioblastoma Multiforme</strong><br />
American Association of Neurological Surgeons<br />
<em>American Association of Neurological Surgeons</em> (2018-08-03) <a href="https://www.aans.org/Patients/Neurosurgical-Conditions-and-Treatments/Glioblastoma-Multiforme">https://www.aans.org/Patients/Neurosurgical-Conditions-and-Treatments/Glioblastoma-Multiforme</a></p>
</div>
<div id="ref-zWyXFgao">
<p>19. <strong>Multiplex methods provide effective integration of multi-omic data in genome-scale models</strong><br />
Claudio Angione, Max Conway, Pietro Lió<br />
<em>BMC Bioinformatics</em> (2016-02) <a href="https://doi.org/gfq976">https://doi.org/gfq976</a><br />
DOI: <a href="https://doi.org/10.1186/s12859-016-0912-1">10.1186/s12859-016-0912-1</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26961692">26961692</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4896256">PMC4896256</a></p>
</div>
<div id="ref-sUdFrajH">
<p>20. <strong>Network-based analysis of omics with multi-objective optimization</strong><br />
Ettore Mosca, Luciano Milanesi<br />
<em>Molecular BioSystems</em> (2013) <a href="https://doi.org/gfq996">https://doi.org/gfq996</a><br />
DOI: <a href="https://doi.org/10.1039/c3mb70327d">10.1039/c3mb70327d</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/24121459">24121459</a></p>
</div>
<div id="ref-dTwzcIZ4">
<p>21. <strong>Gene prioritization through genomic data fusion</strong><br />
Stein Aerts, Diether Lambrechts, Sunit Maity, Peter Van Loo, Bert Coessens, Frederik De Smet, Leon-Charles Tranchevent, Bart De Moor, Peter Marynen, Bassem Hassan, … Yves Moreau<br />
<em>Nature Biotechnology</em> (2006-05) <a href="https://doi.org/bcmjgf">https://doi.org/bcmjgf</a><br />
DOI: <a href="https://doi.org/10.1038/nbt1203">10.1038/nbt1203</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/16680138">16680138</a></p>
</div>
<div id="ref-10RgFYeEF">
<p>22. <strong>Kernel-based data fusion for gene prioritization</strong><br />
Tijl De Bie, Léon-Charles Tranchevent, Liesbeth M. M. van Oeffelen, Yves Moreau<br />
<em>Bioinformatics</em> (2007-07-01) <a href="https://doi.org/b46cvm">https://doi.org/b46cvm</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btm187">10.1093/bioinformatics/btm187</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17646288">17646288</a></p>
</div>
<div id="ref-WkPlH1ds">
<p>23. <strong>Heterogeneous Network Edge Prediction: A Data Integration Approach to Prioritize Disease-Associated Genes</strong><br />
Daniel S. Himmelstein, Sergio E. Baranzini<br />
<em>PLOS Computational Biology</em> (2015-07-09) <a href="https://doi.org/98q">https://doi.org/98q</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004259">10.1371/journal.pcbi.1004259</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26158728">26158728</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4497619">PMC4497619</a></p>
</div>
<div id="ref-O21tn8vf">
<p>24. <strong>Systematic integration of biomedical knowledge prioritizes drugs for repurposing</strong><br />
Daniel Scott Himmelstein, Antoine Lizee, Christine Hessler, Leo Brueggeman, Sabrina L Chen, Dexter Hadley, Ari Green, Pouya Khankhanian, Sergio E Baranzini<br />
<em>eLife</em> (2017-09-22) <a href="https://doi.org/cdfk">https://doi.org/cdfk</a><br />
DOI: <a href="https://doi.org/10.7554/elife.26726">10.7554/elife.26726</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28936969">28936969</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640425">PMC5640425</a></p>
</div>
<div id="ref-Vvpiw3GG">
<p>25. <strong>Network-Based Integration of Disparate Omic Data To Identify “Silent Players” in Cancer</strong><br />
Matthew Ruffalo, Mehmet Koyutürk, Roded Sharan<br />
<em>PLOS Computational Biology</em> (2015-12-18) <a href="https://doi.org/gfq992">https://doi.org/gfq992</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pcbi.1004595">10.1371/journal.pcbi.1004595</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/26683094">26683094</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4684294">PMC4684294</a></p>
</div>
<div id="ref-X7vddlEk">
<p>26. <strong>Synergistic effect of different levels of genomic data for cancer clinical outcome prediction</strong><br />
Dokyoon Kim, Hyunjung Shin, Young Soo Song, Ju Han Kim<br />
<em>Journal of Biomedical Informatics</em> (2012-12) <a href="https://doi.org/f4gstk">https://doi.org/f4gstk</a><br />
DOI: <a href="https://doi.org/10.1016/j.jbi.2012.07.008">10.1016/j.jbi.2012.07.008</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22910106">22910106</a></p>
</div>
<div id="ref-aPrQxP1f">
<p>27. <strong>Graph sharpening plus graph integration: a synergy that improves protein functional classification</strong><br />
Hyunjung Shin, Andreas Martin Lisewski, Olivier Lichtarge<br />
<em>Bioinformatics</em> (2007-10-31) <a href="https://doi.org/dt399h">https://doi.org/dt399h</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btm511">10.1093/bioinformatics/btm511</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/17977886">17977886</a></p>
</div>
<div id="ref-18IBB9pE2">
<p>28. <strong>Time to Recurrence and Survival in Serous Ovarian Tumors Predicted from Integrated Genomic Profiles</strong><br />
Parminder K. Mankoo, Ronglai Shen, Nikolaus Schultz, Douglas A. Levine, Chris Sander<br />
<em>PLoS ONE</em> (2011-11-03) <a href="https://doi.org/c6zcns">https://doi.org/c6zcns</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pone.0024709">10.1371/journal.pone.0024709</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/22073136">22073136</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3207809">PMC3207809</a></p>
</div>
<div id="ref-Hlprh8TG">
<p>29. <strong>Unsupervised Extraction of Stable Expression Signatures from Public Compendia with an Ensemble of Neural Networks</strong><br />
Jie Tan, Georgia Doing, Kimberley A. Lewis, Courtney E. Price, Kathleen M. Chen, Kyle C. Cady, Barret Perchuk, Michael T. Laub, Deborah A. Hogan, Casey S. Greene<br />
<em>Cell Systems</em> (2017-07) <a href="https://doi.org/gcw9f4">https://doi.org/gcw9f4</a><br />
DOI: <a href="https://doi.org/10.1016/j.cels.2017.06.003">10.1016/j.cels.2017.06.003</a> · PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/28711280">28711280</a> · PMCID: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5532071">PMC5532071</a></p>
</div>
</div>
<section class="footnotes">
<hr />
<ol>
<li id="fn1"><p>An oft-included third category is reinforcement learning, in which a computational agent learns actions to maximize a reward function. While such methods show promise for biological applications like the prediction of protein folding, they are beyond the scope of this review.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Note, the Laplacian for a graph is defined as the diagonal degree matrix (elements on the main diagonal correspond to the degree of the respective node) minus the adjacency matrix.<a href="#fnref2" class="footnote-back">↩</a></p></li>
</ol>
</section>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
