## Discussion

Machine learning techniques have broadly been classified into two disjoint categories, supervised and unsupervised [^1].
While in reality the two categories represent endpoints of a continuum, this review will consider the only the binary discretum.
Through such a lens, "supervised" refers to techniques which attempt to learn mappings between specified inputs and outputs.
A simple example of a supervised learning technique is logistic regression, which attempts to assign binary labels to data points after being trained on labeled data.
"Unsupervised" learning refers to techniques which identify characteristics of data without labels.
A simple example of unsupervised learning is principal component analysis (PCA), which transforms observations onto a new set of coordinates that represent the axes of greatest variance.
The discussion that follows is split between supervised and unsupervised methods.

[^1]: An oft-included third category is reinforcement learning, in which a computational agent learns actions to maximize a reward function. While such methods show promise for biological applications like the prediction of protein folding, they are beyond the scope of this review.

### Unsupervised methods

A very useful application of unsupervised methods is the transformation of large noisy data into a useful, de-noised format which can used for supervised machine learning.
For example, Greene et al. [@doi:10.1038/ng.3259] constructed tissue-specific functional gene networks using data from over 14,000 experiments.
By integrating tissue-specific expression information in a Bayesian framework, the authors constructed networks of tissue-specific gene functional relationships, with edge weights representing the posterior probability of a functional relationship.
These edges weights were determined by a naive Bayes classifier that was trained to predict positive functional relationships based on tissue-specific data.
A key insight by the authors was that tissue-specific networks represent as data-driven method for the analysis of noisy GWAS data.
They developed a method called NetWAS to reprioritize genes with nominally-significant p-values from standard GWAS methods.
NetWAS uses the tissue-specific functional networks as features when training a support vector machine (SVM) classifier to predict whether a gene was nominally significant in a GWAS study.
After training, genes can be re-ranked according to their distance from the SVM decision hyperplane.
The rationale behind this method is that the SVM is able to capture some characteristics of the data which lead to significant gene enrichment, while genes with spurious p-values will not share the same functional characteristics.
One of the pivotal innovations of NetWAS is the ability to re-prioritize or cross-prioritize genes from distinct GWAS studies in a method that does not depend on _a priori_ knowledge of the disease.

Another data-integrative application of functional networks was proposed by Sehgal et al. [@doi:10.1371/journal.pone.0140072], who developed a methodology called the Robust Selection Algorithm (RSA) to identify micro-RNAs (miRNA) which are oncogenic or tumor-suppressing.
By incorporating functional networks of miRNA-mRNA data, the authors were able to account for non-linear relationships between molecular markers and patient outcomes.
Using data from TCGA, RSA was able to identify a number of both onco-miRNA and tumor-suppressor miRNA with relevance for the prediction of survival times in various cancers.
To generate functional networks, the authors employed the NetWalker [@doi:10.1186/1471-2164-13-282] suite of software, a self-described, "one-stop platform," for network-based data methods for integrative genomics.

A common goal of network-based systems biology is the identification of functionally-related modules (or sub-networks) within larger networks of genes or proteins.
Ideker et al. [@doi:10.1093/bioinformatics/18.suppl_1.S233] developed an approach to integrate protein-protein and protein-DNA interactions with mRNA expression data to identify "active" subnetworks.
Defined as, "connected sets of genes with unexpectedly high levels of differential expression," active subnetworks were identified by finding groups of nodes with significant differential expression, the _collective_ expression of which was compared to randomly chosen subnetworks.
Having scored subnetworks, the authors' method then finds the highest-scoring subnetworks using simulated annealing.
Subsequent research in module network inference has expanded the methods' capabilities to most common types of omic data.
Lemon-Tree, due to Bonnet et al. [@doi:10.1371/journal.pcbi.1003983], is an open-source software package that encompasses many validated algorithms and novel approaches for module network inference.

iCluster [@doi:10.1093/bioinformatics/btp543] is a joint latent variable model, which attempts to estimate unobserved tumor subtypes using available multiomic data such as CNV, DNA methylation, and mRNA expression.


Network-based methods for biomedical data integration are not limited to representations in which nodes represent genes, proteins, RNA molecules, or other molecular entities.
Similarity Network Fusion (SNF), a method due to Wang, et al. [@doi:10.1038/nmeth.2810], uses patient similarity networks to cluster patients into discrete and clinically-meaningful groups.
In these networks, nodes represent patients and the edges connecting patients represent their pairwise similarity.
For each additional data modality, SNF creates a similarity network based on the patients' molecular profiles.
The constructed modality-specific networks are then fused using an iterative linear algebra routine which is appropriate for studies of varied sample size and feature number, as well as for highly heterogeneous data sources.
The authors applied SNF to three data modalities for a case study of glioblastoma multiforme (GBM), an aggressive brain tumor which is challenging to treat [@url:https://www.aans.org/Patients/Neurosurgical-Conditions-and-Treatments/Glioblastoma-Multiforme].
Previous data-integrative approaches had shown varied results, including the identification of a variable numbers of disease subtypes, depending on the data modality under investigation.
Using SNF, the authors were able to recover known clinically-relevant disease subtypes and show that these subtypes correspond to very different survival prognoses.
Moreover, by probing the network resulting from the SNF method, the authors were able to show that the majority of the edges were supported by two or more different data modalities, strengthening confidence in the validity of their integrative method.

In a separate application of SNF, Angione et al. [@doi:10.1186/s12859-016-0912-1] developed a network approach for bacterial growth and response, in which nodes represent growth conditions and edges represent the similarities between conditions.
In a modification to the original SNF method, Angione et al. propose a weighted or "biased" SNF method in which layers themselves can be weighted according to the predictive quality of the various data.
The authors state that this modification also increases the method's robustness to distributions with high kurtosis.
After consolidating the network into a single, global representation of the multi-modal data, the authors were able to predict _E. coli_ metabolism under a range of possible growth conditions.



### Supervised methods

Kim et al.
[@doi:10.1016/j.jbi.2012.07.008]



[@doi:10.1002/gepi.21628]
[@doi:10.1371/journal.pone.0024709]
[@doi:10.1093/bioinformatics/btt572]
[@doi:10.1093/bioinformatics/bth294]
[@doi:10.1093/bioinformatics/bti1007]
[@doi:10.1093/bioinformatics/bti1110]
[@doi:10.1093/bioinformatics/btm511]

<!-- Review -->
[@doi:10.1038/nrc2294]

HNEP and Rephetio Himmelstein et al., ...
